src/providers/bedrock/chatComplete.ts

```diff
@@ -9,613 +9,218 @@ import {
-                ) {
-                  const parts = item.image_url.url.split(';');
-                  if (parts.length === 2) {
-                    const base64ImageParts = parts[1].split(',');
-                    const base64Image = base64ImageParts[1];
-                    const mediaTypeParts = parts[0].split(':');
-                    if (mediaTypeParts.length === 2 && base64Image) {
-                      const mediaType = mediaTypeParts[1];
-                      transformedMessage.content.push({
-                        type: 'image',
-                        source: {
-                          type: 'base64',
-                          media_type: mediaType,
-                          data: base64Image,
-                        },
-                      });
-                    }
-                  }
-                }
-              });
-              messages.push(transformedMessage as Message);
-            } else if (msg.role === 'tool') {
-              // even though anthropic supports images in tool results, openai doesn't support it yet
-              messages.push(transformToolMessageForAnthropic(msg));
-            } else {
-              messages.push({
-                role: msg.role,
-                content: msg.content,
-              });
-            }
+      transform: (params: BedrockChatCompletionsParams) => {
+        if (!params.messages) return [];
+        return params.messages
+          .filter((msg) => msg.role !== 'system')
+          .map((msg) => {
+            return {
+              role: msg.role === 'assistant' ? 'assistant' : 'user',
+              content: getMessageContent(msg),
+            };
           });
-        }
-
-        return messages;
       },
     },
     {
       param: 'system',
       required: false,
-      transform: (params: Params) => {
-        let systemMessage: string = '';
-        // Transform the chat messages into a simple prompt
-        if (!!params.messages) {
-          params.messages.forEach((msg) => {
-            if (
-              msg.role === 'system' &&
-              msg.content &&
-              typeof msg.content === 'object' &&
-              msg.content[0].text
-            ) {
-              systemMessage = msg.content[0].text;
-            } else if (
-              msg.role === 'system' &&
-              typeof msg.content === 'string'
-            ) {
-              systemMessage = msg.content;
-            }
-          });
-        }
-        return systemMessage;
+      transform: (params: BedrockChatCompletionsParams) => {
+        if (!params.messages) return;
+        const systemMessages = params.messages.reduce(
+          (acc: { text: string }[], msg) => {
+            if (msg.role === 'system')
+              return acc.concat(...getMessageTextContentArray(msg));
+            return acc;
+          },
+          []
+        );
+        if (!systemMessages.length) return;
+        return systemMessages;
+      },
+    },
+    {
```

Comment:
These params should be independently mapped. Currently additionalModelRequestFields and inferenceConfig are added as a part of messages transform logic

Reply:
I think this is the correct approach
InferenceConfig has the following fields
```
          maxTokens: params.max_tokens || params.max_completion_tokens,
          stopSequences:
            typeof params.stop === 'string' ? [params.stop] : params.stop,
          temperature: params.temperature,
          topP: params.top_p,
        };
```
How would you recommend mapping this? If I map it under max_tokens, than would mean that max_tokens has to be present for me to map top_p.

Reply:
All the params should have their individual mapping so that its readable on top-level. All of these can be separate mapping which transforms to `InferenceConfig.*`. You can take a reference from vertex mapping.